### ISSUE TITLE
xiaomi-markw Atmel touchscreen

### ISSUE THREAD

**Issue by 320kg** [msm8916-mainline/lk2nd#428]
Hello! This technically isn't required for Android, but please somehow make markw hardware revisions with the `atmel,maxtouch` touch panel work in the dt. This is required for pmOS. Thanks.

**Comment by TravMurav** [comment]
I believe those devices have multi-sourced touchscreens, in which case it's non-trivial for lk2nd to detect which specific touchscreen is there.

One could probably write a fixup driver and try talking to the touchscreen before boot to figure out which one it is, but this obviously requires someone with the hardware on hand as well as some know-how to correctly detect the present touchscreen.

If I recall correctly, the suggestion for now was to manually build a fixed up verison of lk2nd for a given device.

**Comment by TailsFanLOL** [comment]
Yeah. And I don't speak C good enough yet so I can't do it...

Current workaround is to do this:
```patch
From cbf06795e5d6d1c6f60372dc3b2628fd7e15ca48 Mon Sep 17 00:00:00 2001
From: alex <alekseipesorin@outlook.com>
Date: Sat, 14 Sep 2024 16:54:27 +0300
Subject: [PATCH] atmel

---
 lk2nd/device/dts/msm8953/msm8953-xiaomi-markw.dts | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lk2nd/device/dts/msm8953/msm8953-xiaomi-markw.dts b/lk2nd/device/dts/msm8953/msm8953-xiaomi-markw.dts
index d9d6216..fd14219 100644
--- a/lk2nd/device/dts/msm8953/msm8953-xiaomi-markw.dts
+++ b/lk2nd/device/dts/msm8953/msm8953-xiaomi-markw.dts
@@ -21,7 +21,8 @@
 
 		qcom,mdss_dsi_nt35596_ebbg_1080p_video {
 			compatible = "mdss,nt35596-ebbg";
-			touchscreen-compatible = "edt,edt-ft5336";
+			// touchscreen-compatible = "edt,edt-ft5336";
+			touchscreen-compatible = "atmel,maxtouch";
 		};
 		qcom,mdss_dsi_r63350_1080p_video {
 			compatible = "mdss,r63350";
-- 
2.34.1
```

**Comment by barni2000** [comment]
That is why comments are there to be able to make custom build for atmel variants easily

**Comment by TailsFanLOL** [comment]
Yeah lol

**Comment by ly2101** [comment]
yeah , I have the same problem. so rebuild a lk2nd.img file based on release tag 20.0 source code.
if everyone want to try it .

[postmarketOS-v25.06-25-qcom-msm8953-atmel-lk2nd.zip](https://github.com/user-attachments/files/21362716/postmarketOS-v25.06-25-qcom-msm8953-atmel-lk2nd.zip)

**Comment by barni2000** [comment]
> yeah , I have the same problem. so rebuild a lk2nd.img file based on release tag 20.0 source code.
> if you want to try ,click it 
> 
> [postmarketOS-v25.06-25-qcom-msm8953-atmel-lk2nd.zip](https://github.com/user-attachments/files/21362716/postmarketOS-v25.06-25-qcom-msm8953-atmel-lk2nd.zip)

You don't need to rebuild on 20.0 tag just change the compatible and build it. It is documented on PostmarketOS wiki.
